public/app/features/explore/QueriesDrawer/QueriesDrawerDropdown.tsx

```diff
@@ -1,65 +1,105 @@
 import { css } from '@emotion/css';
+import { ComponentProps, useState } from 'react';
 
+import { config } from '@grafana/runtime';
 import { Button, ButtonGroup, Dropdown, Menu, ToolbarButton } from '@grafana/ui';
 import { useStyles2 } from '@grafana/ui/';
 
+import { createDatasourcesList } from '../../../core/utils/richHistory';
+import { useSelector } from '../../../types';
+import ExploreRunQueryButton from '../ExploreRunQueryButton';
 import { queryLibraryTrackToggle } from '../QueryLibrary/QueryLibraryAnalyticsEvents';
+import { useQueryLibraryContext } from '../QueryLibrary/QueryLibraryContext';
+import { QueryActionButton } from '../QueryLibrary/types';
+import { selectExploreDSMaps } from '../state/selectors';
 
-import { Tabs, useQueriesDrawerContext } from './QueriesDrawerContext';
+import { useQueriesDrawerContext } from './QueriesDrawerContext';
 import { i18n } from './utils';
 
+// This makes TS happy as ExploreRunQueryButton has optional onClick prop while QueryActionButton doesn't
+// in addition to map the rootDatasourceUid prop.
+function ExploreRunQueryButtonWrapper(props: ComponentProps<QueryActionButton>) {
+  return <ExploreRunQueryButton {...props} rootDatasourceUid={props.datasourceUid} />;
+}
+
 type Props = {
   variant: 'compact' | 'full';
 };
 
+/**
+ * Dropdown button that can either open a Query History drawer or a Query Library drawer.
+ * @param variant
+ * @constructor
+ */
 export function QueriesDrawerDropdown({ variant }: Props) {
-  const { selectedTab, setSelectedTab, queryLibraryAvailable, drawerOpened, setDrawerOpened } =
-    useQueriesDrawerContext();
+  const { drawerOpened, setDrawerOpened } = useQueriesDrawerContext();
+
+  const {
+    openDrawer: openQueryLibraryDrawer,
+    closeDrawer: closeQueryLibraryDrawer,
+    isDrawerOpen: isQueryLibraryDrawerOpen,
+  } = useQueryLibraryContext();
+
+  const [queryOption, setQueryOption] = useState<'library' | 'history'>('library');
+
+  const exploreActiveDS = useSelector(selectExploreDSMaps);
 
   const styles = useStyles2(getStyles);
 
-  if (!queryLibraryAvailable) {
+  // In case query library is not enabled we show only simple button for query history in the parent.
+  if (!config.featureToggles.queryLibrary) {
     return undefined;
   }
 
-  function toggle(tab: Tabs) {
-    tab === Tabs.QueryLibrary && queryLibraryTrackToggle(!drawerOpened);
+  function toggleRichHistory() {
+    setQueryOption('history');
+    setDrawerOpened(!drawerOpened);
+  }
+
+  function toggleQueryLibrary() {
```

Comment:
This mainly changes some of the handling of the toggles as we have 2 separate contexts and states now instead of only one.